bio2zarr/tskit.py

```diff
@@ -143,13 +149,12 @@ def iter_alleles_and_genotypes(self, start, stop, shape, num_alleles):
                 assert i < num_alleles
                 alleles[i] = allele
 
-            genotypes = variant.genotypes
-            sample_index = 0
-            for i, ploidy in enumerate(self.individual_ploidies):
-                for j in range(ploidy):
-                    if j < self.max_ploidy:
-                        gt[i, j] = genotypes[sample_index + j]
-                sample_index += ploidy
+            # For each individual, get genotypes for their nodes
+            for i in range(self.num_samples):
```

Comment:
There must be some numpy-way of doings this without iterating over samples - otherwise this is going to be super slow